src/main/java/org/entando/plugins/pda/pam/service/task/KieTaskFormService.java

```diff
@@ -64,21 +61,17 @@
     }
 
     @Override
-    public Task submit(Connection connection, AuthenticatedUser user, String id,
-            CreateTaskFormSubmissionRequest request) {
-
-        KieInstanceId taskId = new KieInstanceId(id);
+    public String submit(Connection connection, AuthenticatedUser user, String id, Map<String, Object> request) {
+        Form form = get(connection, id);
 
-        Map<String, Object> variables = new ConcurrentHashMap<>();
-        Optional.ofNullable(request.getForms())
-                .orElseThrow(BadRequestException::new)
-                .values().forEach(variables::putAll);
+        Map<String, Object> variables = createFormSubmission(form, request);
 
         UserTaskServicesClient client = kieApiService.getUserTaskServicesClient(connection);
 
         try {
+            KieInstanceId taskId = new KieInstanceId(id);
             client.saveTaskContent(taskId.getContainerId(), taskId.getInstanceId(), variables);
-            return taskService.get(connection, user, id);
+            return id;
```

Comment:
Because of this change, `taskService` attribute is not required anymore and can be removed from this class.